src/main.ts

```diff
@@ -381,23 +247,33 @@ function resetTrackUI(): void {
 
 const setSize = (): void => canv.setSize(Math.min(window.innerWidth, 730), Math.min(window.innerHeight, 500));
 
-async function drawFrame(index: number): Promise<void> {
-  await canv.setFrame(index);
-  canv.render();
-}
-
 async function drawLoop(): Promise<void> {
   if (dataset && datasetOpen) {
     // update higlighted cell id if any
     if (selectedTrack) {
-      const id = selectedTrack.getIdAtTime(timeControls.getCurrentFrame());
+      const id = selectedTrack.getIdAtTime(canv.getCurrentFrame());
       canv.setHighlightedId(id - 1);
-      // console.log(`selected track: ${selectedTrack.trackId}; highlighted id ${id}`);
     }
-    // update current time in plot
-    plot.setTime(timeControls.getCurrentFrame());
-    await drawFrame(timeControls.getCurrentFrame());
   }
+
+  await canv.render();
```

Comment:
this seems like the place to do the file saving if we are in a recording state.

Reply:
Also when thinking about dropped frames, I believe maybe we should be using requestAnimationFrame in this app to just run an infinite redraw loop.

Reply:
@ShrimpCryptid and I discussed a few ways to make this better and the result of that conversation is #30 but maybe not necessary to implement in this PR unless @frasercl thinks so...  We also agreed on an improvement that will make setFrame not have to return false.